mxcubeweb/core/server/openapidoc.py

```diff
@@ -0,0 +1,217 @@
+import re
+
+from flask import Blueprint, jsonify
+from pydantic.v1 import (
+    BaseModel,
+)
+
+DEFAULT_RESPONSES = {
+    "200": {"description": "Success"},
+    "400": {"description": "Invalid input data"},
+    "404": {"description": "Invalid or non existing object"},
+    "500": {"description": "Error calling method on adapter"},
+}
+
+
+def to_openapi_path(route: str) -> str:
+    """
+    Adds the "{" "}" to the route arguments that OpenAPI requires
+
+    Args:
+        route(str): The route
+
+    Returns:
+        returns: the openapi path, with the {} srounding the arguments
+    """
+    _s = re.sub(r"<[^:]+:", "{", route)
+
+    # Replace closing ">" with "}"
+    return _s.replace(">", "}")
+
+
+class OpenAPISpec:
```

Comment:
We are also replacing the unmaintained `spectree` library with this fairly simple class `OpenAPISpec`